Project/Assets/ML-Agents/Examples/SharedAssets/Scripts/DirectionIndicator.cs

```diff
@@ -1,27 +1,41 @@
-﻿using System.Collections;
-using System.Collections.Generic;
-using UnityEngine;
+﻿using UnityEngine;
 
 namespace Unity.MLAgentsExamples
 {
     public class DirectionIndicator : MonoBehaviour
     {
+        public bool updateViaScript;
         public Transform transformToFollow; //ex: hips or body
         public Transform targetToLookAt; //target in the scene the indicator will point to
```

Comment:
Both the DirectionIndicator and OrientationCubeController are changing to allow an agent to update the scripts. Not all environments have been changed/updated yet to this new pattern so some still need public fields to add references.

Reply:
I am confused, `transformToFollow` is not used anywhere in the code except in `DirectionIndicator.cs`. Therefore, I think it should be made private.
Your comment is not 100% clear to me, what do you mean by 
```

Reply:
the reason for having it public is to make easier to configure prefabs and access this property during runtime if needed. 

regarding the confusing comment... before this script couldn't be updated by the agent since there were no public methods exposed. now the agent can update it.  i changed the bool "updateViaScript" to "updatedByAgent" and added comments to make the intended purpos clearer.

Reply:
Why not make it private and use [SerializeField] attribute so it is modifiable from the inspector and prefabs?